ansys/mapdl/core/license.py

```diff
@@ -0,0 +1,349 @@
+"""
+Everything related to licensing and license serve checks. 
+
+"""
+
+import os 
+import time 
+import warnings
+import re 
+import socket 
+import subprocess
+import datetime
+
+
+from ansys.mapdl.core.errors import LicenseServerConnectionError  
+
+
+LOCALHOST = '127.0.0.1'
+
+
+def try_license_file():
+    
+    try:
+        license_file_checker()
+    except LicenseServerConnectionError:
+        # expected error, so let it raise
+        raise 
+    except:
+        # Rest of the cases, let's run secondary methods.
+        # print("Error found trying to read license file.")
+        return False
+    else:
+        return True 
+
+
+def license_file_checker():
+    licdebug_file = os.path.join(get_licdebug_path(), get_licdebug_name())
+    file_iterator = get_licdebug_msg(licdebug_file)
+
+    time_to_be_checking = 10 #seconds
+    max_time = datetime.time(0, 0, time_to_be_checking) #hours, minute, seconds
+
+    while max_time < datetime.datetime.now().time():
+        msg = next(file_iterator)
+
+        if is_denied_msg(msg):
+            license_path = re.findall( "(?<=License path:)(.*)(?=;\n)", msg)[0]
+            license_port = license_path.split('@')[0]
+            license_hostname = license_path.split('@')[1]
+            raise LicenseServerConnectionError(
+                head_message=f"Error connecting to {license_port}:{license_hostname}",
+                error_message=msg,
+                tail_message=f"Error found in file {get_licdebug_name()}")
+        
+    else:
+        print("Time is out for license checking.")
```

Comment:
```suggestion
        else:
            print("Time is out for license checking.")
```

Reply:
Actually, the `else` is for the `while`.

Reply:
Okay I see. My mistake! Ignore this one then.